### ISSUE TITLE
Add Hotkeys for changing subtitle offset

### ISSUE THREAD

**Issue by tstrul1** [carsonip/Penguin-Subtitle-Player#10]
It could be really useful if you could add this feature. Maybe something like:
Ctrl + -> and Ctrl + <- for changing subtitle offset while watching movie

**Comment by carsonip** [comment]
@tstrul1 thanks for the suggestion, will work on it soon.

**Comment by tstrul1** [comment]
I guess you stopped developing this player? :\

**Comment by carsonip** [comment]
@tstrul1 I'm really sorry for not implementing this new feature right away. The project is currently not under active development but there would still be bug fixes. Pull requests are welcomed! Of course I'll implement new features when I have the time. Tho the problem with this feature is that when the player is not focused, we will probably not receive any events and we may need a keylogger here (which will be quite troublesome).

**Comment by neopostmodern** [comment]
If someone is interested in getting this to work (on X11 systems), I've hacked together something similar for next/previous: https://github.com/neopostmodern/Penguin-Subtitle-Player/commit/3424b29a6853eb0ef86a79b3d9decfc9755058ad
I don't think it's suitable for a PR though, because it's X11-only...

**Comment by carsonip** [comment]
> If someone is interested in getting this to work (on X11 systems), I've hacked together something similar for next/previous: [neopostmodern@3424b29](https://github.com/neopostmodern/Penguin-Subtitle-Player/commit/3424b29a6853eb0ef86a79b3d9decfc9755058ad) I don't think it's suitable for a PR though, because it's X11-only...

Hi @neopostmodern , thanks for working on it! I'm starting to think that a feature like this does not have to work on all systems at a time. As long as we figure out a way to isolate that code and keep other features working on other systems, it may be good enough. However if it results in multiple compiled versions and releases, it will make releasing quite troublesome.

**Comment by neopostmodern** [comment]
Yeah, I guess it would become a pain. Because even if you conditionally included this for Linux systems only it would break for the Wayland folks, no? I don't see any easy way out of this, other than finding some general Qt-abstraction for hotkey capture...
You could build one yourself, but I'd think that it somehow should exist already?

**Comment by carsonip** [comment]
> Yeah, I guess it would become a pain. Because even if you conditionally included this for Linux systems only it would break for the Wayland folks, no?

Possibly, I guess? Not too familiar with this.

> I don't see any easy way out of this, other than finding some general Qt-abstraction for hotkey capture... You could build one yourself, but I'd think that it somehow should exist already?

A quick google search returns [qxtglobalshortcut](https://github.com/hluk/qxtglobalshortcut) but I didn't look into it further.
